app/controllers/ChoiceController.scala

```diff
@@ -31,15 +32,24 @@ import views.html.choice_page
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ChoiceController @Inject()(authenticate: AuthAction, cache: CacheRepository, mcc: MessagesControllerComponents, choicePage: choice_page)(
-  implicit ec: ExecutionContext
-) extends FrontendController(mcc) with I18nSupport {
+class ChoiceController @Inject()(
+  authenticate: AuthAction,
+  cacheRepository: CacheRepository,
+  mcc: MessagesControllerComponents,
+  choicePage: choice_page
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
 
-  def displayChoiceForm: Action[AnyContent] = authenticate.async { implicit request =>
-    cache.findByEori(request.eori).map(_.flatMap(_.answers)).map {
-      case Some(answers) => Ok(choicePage(Choice.form().fill(Choice(answers.`type`))))
-      case None          => Ok(choicePage(Choice.form()))
-    }
+  def displayChoiceForm(): Action[AnyContent] = authenticate.async { implicit request =>
+    cacheRepository
+      .findByEori(request.eori)
+      .map {
+        case Some(cache) =>
+          cache.answers
+            .map(answers => Ok(choicePage(Choice.form().fill(Choice(answers.`type`)))))
+            .getOrElse(Ok(choicePage(Choice.form())))
```

Comment:
Actually, I think you should be able to do deep pattern matching here:
`case Some(Cache(_, Some(answers), _, _)) => Ok(choicePage(Choice.form().fill(Choice(answers.`type`))))`

Reply:
Nevermind, you solution will be handy once redirection for no cache is implemented.

Reply:
There was no need to change this method.  As indicated in your TODO, I've done/am doing this as part of 2175